lib/host/events/resubmit_all_orders.js

```diff
@@ -0,0 +1,46 @@
+'use strict'
+
+const _isObject = require('lodash/isObject')
+const withAOUpdate = require('../with_ao_update')
+
+/**
+ * Re-submits all provided orders with the specified delay, and adds them to the
+ * AO instance state. Attaches meta labels if the algo order supports them.
+ *
+ * @param {object} aoHost - algo host
+ * @param {string} gid - AO instance gid
+ * @param {object[]|Array[]} orders - orders to be submitted
+ * @param {number} delay - cancellation delay
+ * @returns {object} nextInstanceState
+ */
+module.exports = async (aoHost, gid, orders, delay) => {
+  await withAOUpdate(aoHost, gid, async (instance = {}) => {
+    const { state = {}, h = {} } = instance
+    const { id } = state
+    const { submitOrderWithDelay, debug } = h
+    const _orders = _isObject(orders)
+      ? Object.values(orders).filter(order => order.status === 'REQUEUE')
+      : orders.filter(order => order.status === 'REQUEUE')
```

Comment:
odd that the internal format for the data changes sometimes... its sometimes an object and sometimes an array?

Reply:
hmm.. depends on the format you send it.. its just like a precaution so that it works for both cases.. there are formats where you save the orders in key value format where key is the ocid assigned to the order and value being the order object.. otherwise the orders are sent directly in the array format